portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -14,34 +14,72 @@
 
 package com.google.sps.servlets;
 
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.Query.SortDirection;
 import com.google.gson.Gson;
 import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Arrays;
 import javax.servlet.annotation.WebServlet;
 import javax.servlet.http.HttpServlet;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
-import java.util.ArrayList;
-import java.util.Arrays;
 
 /** Servlet that returns some example content. TODO: modify this file to handle comments data */
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
   ArrayList<String> messages = new ArrayList<>();
-  @Override
-  public void init(){
-      messages.add("Spotify or Apple Music");
-      messages.add("Light mode or dark mode?");
-      messages.add("Mac or PC?");
-  }
+  ArrayList<CommentData> comments = new ArrayList<>(); 
+
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    String json_messages = convertToJson(messages);
+    Query query = new Query("Comment");
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    PreparedQuery results = datastore.prepare(query);
+    for (Entity entity : results.asIterable()){
+        String name = (String) entity.getProperty("name");
+        String email = (String) entity.getProperty("email");
+        String message = (String) entity.getProperty("message");
+        CommentData comment = new CommentData(name,email,message);
+        comments.add(comment);
+    }
+    String json_messages = convertToJson(comments);
     response.setContentType("application/json;");
     response.getWriter().println(json_messages);
   }
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+      String name = request.getParameter("full-name");
+      String email = request.getParameter("email");
+      String message = request.getParameter("message");
+
+      Entity commentEntity = new Entity("Comment");
+      commentEntity.setProperty("name",name);
+      commentEntity.setProperty("email",email);
+      commentEntity.setProperty("message",message);
+
+      DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+      datastore.put(commentEntity);
+
+      response.sendRedirect("/index.html");
+  }
   private String convertToJson(ArrayList items){
       Gson gson = new Gson();
       String json = gson.toJson(items);
       return json;
   }
+  public class CommentData {
```

Comment:
Could CommentData be private (or at least not public) since it looks like it's only being used within DataServlet?